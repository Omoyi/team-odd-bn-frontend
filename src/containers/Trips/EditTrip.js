import React, { Component } from 'react';
// import { connect } from 'react-redux';
import '../../assets/css/editTrip.css';



class EditRequest extends Component {
  render() {
    return (
      <>

          <form>
            <input type="submit" value="Save Changes" />
            <br />
            <label >Origin</label>
            <input type="text" id="origin" name="origin" placeholder="Your Origin.." />

            <label>Destination</label>
            <input type="text" id="destination" name="ldestination" placeholder="Your Destination.." />

            <label>Start Date</label>
            <input type="text" id="StartDate" name="StartDate" placeholder="Your Start Date.." />

            <label>Return Date</label>
            <input type="text" id="Return Date" name="lReturn Date" placeholder="Your Return Date.." />

            <label>Reason</label>
            <input type="text" id="Reason" name="Reason" placeholder="Your Reason.." />


          </form>

      </>
    );
  }
}


export default EditRequest;
