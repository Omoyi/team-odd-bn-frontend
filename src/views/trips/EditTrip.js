import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../assets/css/table.scss';
import Pagination from 'custom_react_pages';
import {
  fetchRequestsAction,
  editRequestAction,
  paginationAction,
  editTripsAction,
} from '../../redux/actions/tripsActions/fetchRequests';
import popUpAction from '../../redux/actions/popUpAction';
import { getTripAction } from '../../redux/actions/tripsActions/onewayActions';
import { changePageNo } from '../../redux/actions/PaginationAction';
import prev from '../../assets/images/prev_icon.png';
import next from '../../assets/images/next_icon.png';

class Requests extends Component {
  async componentDidMount() {
    await this.props.getTripAction();
  }

  render() {
    const {
      singleRequestData,
      editTripsAction,
      cities,
      editRequestAction,
      id,
      changePageNo,
      fetchRequestsAction,
      pageNo,
    } = this.props;
    const { data } = singleRequestData;
    const { trips, ...rest } = data || { trips: [] };

    return (
      <div className="editPopupContainer">

        {data !== undefined && (
          <Pagination
            activePageStyle={{ backgroundColor: '#00b9f2', color: 'white' }}
            itemsPerPage={1}
            next={<img src={next} />}
            prev={<img src={prev} />}
            data={trips || []}
            pageName="Trip"
            pageButtons={10}
            onePage={(item, index) => (
              <div key={index} className="inputFields">
                <div className="saveEditedRequests">
                  <button
                    type="button"
                    id="cancel"
                    onClick={() => {
                      this.props.popUpAction({
                        currentPopUp: 'editTrip',
                        editTrip: 'none',
                      });
                    }}
                  >
            &times;
                  </button>
                </div>
                <>
                  <div className="editRequestFields">
                    <div className="label">Origin</div>
                    <div>
                      <select
                        value={item.originId}
                        name="originId"
                        id="originId"
                        onChange={(e) => trips && editTripsAction(
                          { data: trips },
                          {
                            tripIndex: pageNo,
                            name: e.target.name,
                            value: e.target.value,
                          },
                          rest,
                        )}
                      >
                        {cities
                    && cities.map((city, index) => (
                      <option name="originId" key={city.id} value={city.id}>
                        {city.city}
                      </option>
                    ))}
                      </select>
                    </div>
                  </div>
                  <div key={`${index}nd`} className="editRequestFields">
                    <div className="label">Destination</div>
                    <div>
                      <select
                        value={item.destinationId}
                        name="destinationId"
                        id="destinationId"
                        onChange={(e) => editTripsAction(
                          { data: trips },
                          {
                            tripIndex: pageNo,
                            name: e.target.name,
                            value: e.target.value,
                          },
                          rest,
                        )}
                      >
                        {cities
                      && cities.map((city) => (
                        <option
                          name="destinationId"
                          key={city.id}
                          value={city.id}
                        >
                          {city.city}
                        </option>
                      ))}
                      </select>
                    </div>
                  </div>
                  <div key={`${index}nth`} className="editRequestFields">
                    <div className="label">Start date</div>
                    <div>
                      <input
                        type="date"
                        id="startDate"
                        name="startDate"
                        value={item.startDate}
                        onChange={(e) => editTripsAction(
                          { data: trips },
                          {
                            tripIndex: pageNo,
                            name: e.target.name,
                            value: e.target.value,
                          },
                          rest,
                        )}
                      />
                    </div>
                  </div>
                  <div key={`${index}rd`} className="editRequestFields">
                    <div className="label">Return date</div>
                    <div>
                      <input
                        type="date"
                        onChange={(e) => editTripsAction(
                          { data: trips },
                          {
                            tripIndex: pageNo,
                            name: e.target.name,
                            value: e.target.value,
                          },
                          rest,
                        )}
                        id="returnDate"
                        name="returnDate"
                        value={item.returnDate}
                      />
                    </div>
                  </div>
                  <div key={`${index}th`} className="editRequestFields">
                    <div className="label">reason</div>
                    <div>
                      <textarea
                        type="text"
                        onChange={(e) => editTripsAction(
                          { data: trips },
                          {
                            tripIndex: pageNo,
                            name: e.target.name,
                            value: e.target.value,
                          },
                          rest,
                        )}
                        name="reason"
                        value={item.reason}
                      />
                    </div>
                  </div>
                </>
                <div className="saveEditedRequests">
                  <button
                    type="button"
                    id="save"
                    onClick={async () => {
                      trips && await editRequestAction(id, { itinerary: trips });
                      await fetchRequestsAction(id);
                      this.props.popUpAction({
                        currentPopUp: 'editTrip',
                        editTrip: 'none',
                      });
                    }}
                  >
            save changes
                  </button>
                </div>
              </div>

            )}
          />
        )}
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  singleRequestData: state.trips.requests.singleRequestData,
  cities: state.trips.tripRequests.getCity,
  pageNo: state.pagination.pageNo,
  itemsPerPage: state.pagination.itemsPerPage,
});
const actions = {
  fetchRequestsAction,
  paginationAction,
  popUpAction,
  editTripsAction,
  getTripAction,
  editRequestAction,
  changePageNo,
};

export default connect(mapStateToProps, actions)(Requests);
