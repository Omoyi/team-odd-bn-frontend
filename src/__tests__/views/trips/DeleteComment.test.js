import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import EditTrip from '../../../views/trips/DeleteComment';
import {
  init,
} from '../../../__mocks__/trips/requests';

const mockStore = configureStore([]);
let wrapper;
let store;

describe('Test Signup page ', () => {
  it('Test snapshot using the initial state', () => {
    store = mockStore(init);
    wrapper = mount(<Provider store={store}><Router><EditTrip store={store} singleRequestData={{ singleRequestData: { data: { trips: [] } } }} /></Router></Provider>);


    const cancel = wrapper.find('#cancel');
    const deleteMessage = wrapper.find('#delete');
    cancel.simulate('click');
    deleteMessage.simulate('click');
    cancel.simulate('click');
    expect(jest.fn().mock.calls).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
