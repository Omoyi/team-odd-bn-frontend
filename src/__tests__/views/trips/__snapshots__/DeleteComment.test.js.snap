// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Signup page  Test snapshot using the initial state 1`] = `ReactWrapper {}`;
